feat: display underlying raw data for aggregated pivot cell

djbarnwal opened the pull request:
Closes #4328

ericpgreen2 commented:
Sweet! Probably merits a tooltip & a visual indication for the active cell. Maybe @jkhwu has input.

djbarnwal commented:
Addressed the review comments

>     * if the pivot table is empty, rather than showing any data, I think I'd expect the row viewer to say something like "no active cell"

For this do we want to only show data when we have an active cell or even if we have a valid pivot table but no active cell.

nishantmonu51 commented:
When we have a valid pivot table and no active cell, we can show the data for all cells by just applying global filters. It would be consistent with how it works in explore as well. In order to make it clear visually indicate the difference in the table header.

ericpgreen2 commented:
Styling nits:
- There should still be an indicator for the active cell when the row as a whole is highlighted. In the following pic, after clicking on the "Above the fold > Total Records" cell, the active cell highlight didn't show up because the row highlight was overpowering it.
    <img width="1281" alt="image" src="https://github.com/user-attachments/assets/79dcc578-e595-40f9-b705-e3f3c4563aed">
- To indicate what the active cell will be, we should highlight the cell on hover (in addition to the row as a whole).

djbarnwal commented:
Made the changes

With active cell
![image](https://github.com/user-attachments/assets/6ce09894-3dc3-4fa8-9b86-d72d80ecdfba)

Without active cell or empty pivot
![image](https://github.com/user-attachments/assets/89a8dd69-9dbc-4fd0-9bdb-d7c602386f41)

For explore, the header is same as before

ericpgreen2 commented:
Nice, the on:hover styling feels great now!

Some more QA:
1. I'm not seeing the same cell interactions in Rill Cloud as I am in Rill Developer. (For a project where I've enabled the `cloudDataViewer`.)
2. It seems we're too aggressively clearing the active cell state. For example, when I click on a cell, drag the Model Data section up/down, the active cell is lost.
3. When the Model Data section is open & dragged to a custom height, I wouldn't expect that clicking on another active cell would reset the height to its default.
4. It's not totally clear that "for selected global filters (and highlighted pivot cell)" refers to the numerator not the denominator. IMO we should remove that italicized text and instead reconsider the "Model Data" copy.

ericpgreen2 commented on web-common/src/features/dashboards/rows-viewer/RowsViewerAccordion.svelte:
```diff
@@ -1,54 +1,99 @@
+    }
+    filters = sanitiseExpression(activeCellFilters.filters, undefined);
+    timeRange = activeCellFilters.timeRange;
+    label = "for selected global filters and highlited pivot cell";
```
"highlited" -> "highlighted"

ericpgreen2 commented on web-common/src/features/dashboards/rows-viewer/row-viewer-store.ts:
```diff
@@ -0,0 +1,3 @@
+import { writable } from "svelte/store";
+
+export const rowViewerStore = writable<HTMLElement | null>(null);
```
Mind just adding a comment to explain why this is needed?

ericpgreen2 reviewed (approved):
Looks good! Approving.

Just the one thing I still think should be reconsidered is the "for selected global filters (and highlighted pivot cell)" text. @ericokuma @mindspank @jkhwu any feedback?

djbarnwal commented:
Synced up with @jkhwu and we decided to use `Model Data` and `Model data for selected cell` terminology when a cell is highlighted.
